docs/standard/microservices-architecture/implement-resilient-applications/monitor-app-health.md

```diff
@@ -15,146 +15,236 @@ In the typical model, services send reports about their status, and that informa
+    return services;
+}
 
-By default, the cache duration is internally set to 5 minutes, but you can change that cache duration on each health check, as in the following code:
+```
+Finally, we add the HealthCheck middleware to listen to “/hc” endpoint:
 
 ```csharp
-checks.AddUrlCheck(Configuration["CatalogUrl"],1); // 1 min as cache duration
+// HealthCheck middleware
+app.UseHealthChecks("/hc", new HealthCheckOptions()
+{
+    Predicate = _ => true,
+    ResponseWriter = UIResponseWriter.WriteHealthCheckUIResponse
+});
+}
 ```
 
 ### Querying your microservices to report about their health status
 
-When you have configured health checks as described here, once the microservice is running in Docker, you can directly check from a browser if it is healthy. (This does require that you are publishing the container port out of the Docker host, so you can access the container through localhost or through the external Docker host IP.) Figure 10-7 shows a request in a browser and the corresponding response.
+When you’ve configured health checks as described in this article and you have the microservice running in Docker, you can directly check from a browser if it’s healthy. You have to publish the container port in the Docker host, so you can access the container through the external Docker host IP or through localhost, as shown in figure 10-7.
+
 
 ![](./media/image7.png)
 
 **Figure 10-7**. Checking health status of a single service from a browser
 
-In that test, you can see that the catalog.api microservice (running on port 5101) is healthy, returning HTTP status 200 and status information in JSON. It also means that internally the service also checked the health of its SQL Server database dependency and that health check was reported itself as healthy.
+In that test, you can see that the Catalog.API microservice (running on port 5101) is healthy, returning HTTP status 200 and status information in JSON. The service also checked the health of its SQL Server database dependency and RabbitMQ thus the health check reported itself as healthy.
 
 ## Using watchdogs
 
 A watchdog is a separate service that can watch health and load across services, and report health about the microservices by querying with the HealthChecks library introduced earlier. This can help prevent errors that would not be detected based on the view of a single service. Watchdogs also are a good place to host code that can perform remediation actions for known conditions without user interaction.
 
 The eShopOnContainers sample contains a web page that displays sample health check reports, as shown in Figure 10-8. This is the simplest watchdog you could have, since all it does is shows the state of the microservices and web applications in eShopOnContainers. Usually a watchdog also takes actions when it detects unhealthy states.
```

Comment:
```suggestion
The eShopOnContainers sample contains a web page that displays sample health check reports, as shown in Figure 10-8. This is the simplest watchdog you could have, since all it does is show the state of the microservices and web applications in eShopOnContainers. Usually, a watchdog also takes actions when it detects unhealthy states.
```